write.go

```diff
@@ -177,8 +169,10 @@ func (w *Writer) writeField(field interface{}) error {
 		return w.writeLiteral(field)
 	case []interface{}:
 		return w.writeList(field)
+	case *timeLayout:
```

Comment:
1. Extensions in other packages should be able to specify in which format they want dates to be formatted.
2. Maybe something like `Quoted` would be better? The idea is to add:

   ```go
   type Date time.Time
   type DateTime time.Time
   ```

   And use the corresponding format in this switch. Another solution would be to add some exported functions to create `timeLayout`s.